tests/functional/conftest.py

```diff
@@ -1,100 +1,344 @@
     ClientCredentialsConfig,
     ClientCredentialsAuthenticator,
 )
 
 from pytest_nhsd_apim.apigee_apis import (
-    ApigeeClient,
     ApiProductsAPI,
-    ApigeeNonProdCredentials,
     DeveloperAppsAPI,
 )
 
-import logging
-
-LOGGER = logging.getLogger(__name__)
-DEVELOPER_EMAIL = "apm-testing-internal-dev@nhs.net"
-
-
-@pytest.fixture()
-def client():
-    config = ApigeeNonProdCredentials()
-    return ApigeeClient(config=config)
+FILE_DIR = os.path.dirname(__file__)
+RESPONSES_DIR = os.path.join(FILE_DIR, 'data', 'responses')
 
 
 @pytest.fixture()
-def api_products(client):
-    return ApiProductsAPI(client=client)
+def developer_email() -> str:
+    return "apm-testing-internal-dev@nhs.net"
 
 
 @pytest.fixture()
-def developer_apps(client):
-    return DeveloperAppsAPI(client=client)
+def pds_url() -> str:
+    return f"{BASE_URL}/{PDS_BASE_PATH}"
 
 
 @pytest.fixture()
-def add_asid_to_testapp(developer_apps, nhsd_apim_test_app):
+def add_asid_to_testapp(developer_apps_api,
+                        nhsd_apim_test_app,
+                        developer_email):
     app = nhsd_apim_test_app()
-    LOGGER.info(f'app:{app}')
     app_name = app["name"]
 
-    # Check if the ASID attribute is already available
-    app_attributes = developer_apps.get_app_attributes(email=DEVELOPER_EMAIL, app_name=app_name)
+    app_attributes = developer_apps_api.get_app_attributes(email=developer_email, app_name=app_name)
     custom_attributes = app_attributes['attribute']
     existing_asid_attribute = None
     for attribute in custom_attributes:
         if attribute['name'] == 'asid':
             existing_asid_attribute = attribute['value']
 
     if not existing_asid_attribute and config.ENV.contains("internal_dev_asid"):
-        LOGGER.info(f'ASID attribute not found. Adding {config.ENV["internal_dev_asid"]} to {app_name}')
-        # Add ASID to the test app - To be refactored when we move to .feature files TODO SPINEDEM-1680
         custom_attributes.append({"name": "asid", "value": config.ENV["internal_dev_asid"]})
         data = {"attribute": custom_attributes}
-        response = developer_apps.post_app_attributes(email=DEVELOPER_EMAIL, app_name=app_name, body=data)
-        LOGGER.info(f'Test app updated with ASID attribute: {response}')
+        developer_apps_api.post_app_attributes(email=developer_email, app_name=app_name, body=data)
+
+
+@given("I am an unknown user", target_fixture='headers_with_authorization')
+def provide_headers_with_no_auth_details() -> None:
+    return {}
+
+
+@given("I am a healthcare worker")
+def provide_healthcare_worker_auth_details(request) -> None:
+    auth_details = {
+        "api_name": "personal-demographics-service",
+        "access": "healthcare_worker",
+        "level": "aal3",
+        "login_form": {"username": "656005750104"},
+        "force_new_token": True
+    }
+    request.node.add_marker(pytest.mark.nhsd_apim_authorization(auth_details))
+
```

Comment:
so nice to read code with snake_casing! if only all repos were written like this!